pkg/api/types.go

```diff
@@ -2607,6 +2607,8 @@ type ReplicationControllerCondition struct {
 }
 
 // +genclient
+// +genclient:method=GetScale,verb=get,subresource=scale,result=k8s.io/kubernetes/pkg/apis/extensions.Scale
```

Comment:
Why not treating the Scale subresource the same way we treat Status subresource? We defined templates for UpdateStatus and optionally generated the `UpdateStatus()` methods.

Are there other use cases for the `method` UI? It's useful to mass generating regular CRUD operations for subresources, but for that use case, the approach we took for "Status" is easier to understand.

Reply:
@mfojtik btw, it would be great if you can update the client-gen [doc](https://github.com/kubernetes/community/blob/master/contributors/devel/generating-clientset.md) for the last major improvement you did for client-gen :)

Reply:
> the approach we took for "Status" is easier to understand.

That approach doesn't work for scale. The type passed to scale is not the same as the main resource, and the field name that indicates you can scale varies

Reply:
@caesarxuchao about use-cases for `method`: this is for thirdparty apiservers and their api groups. They might have more (and different) subresources than we have in kube with mainly `/status` and `/scale`. This PR enables them to use client-gen effectively. I would prefer a general method like the proposed one than people re-implementing plenty of expansion methods. The former are declarative and stable, the later will break if we change an interface somewhere, e.g. the rest client.

Reply:
@caesarxuchao i don't think the way we generate Status is good TBH. We rely on the presence of `.Status` field and if that field exists we generate the `UpdateStatus()`, but in case the `.Status` field exists but the `status` subresource don't you need to indicate that with tag...  tbh, I don't like that sort of magic :-)

I believe that using `// +genclient:method=UpdateStatus,subresource=status` is more explicit. Not super-easy to use but leads to cleaner generation :-) WDYT? // cc @sttts

Reply:
100% :+1: for @mfojtik's argument with "that sort of magic" regarding the `/status` subresource.

Reply:
Thanks for the explanation. The thirdparty use cases and explicit tags for `UpdateStatus` sound good.